cdmtaskservice/mongo.py

```diff
@@ -105,6 +110,17 @@ async def get_image(self, name: str, digest: str | None = None, tag: str | None
             raise NoSuchImageError(f"No image {name} {err} exists in the system.")
         return models.Image.model_construct(**doc)
 
+    async def save_job(self, job: models.Job):
+        """ Save a job. Job IDs are expected to be unique."""
+        _not_falsy(job, "job")
+        jobd = job.model_dump()
+        jobi = jobd[models.FLD_JOB_JOB_INPUT]
+        jobi[models.FLD_JOB_INPUT_RUNTIME] = jobi[models.FLD_JOB_INPUT_RUNTIME].total_seconds()
+        # don't bother checking for duplicate key exceptions since the service is supposed
+        # to ensure unique IDs
+        # TODO test that the timedelta is correctly reinstantiated in a get_job method
+        await self._col_jobs.insert_one(jobd)
```

Comment:
Not sure if it's worth to add a `try-catch` block here. DB client sometimes can generate error logs that are difficult to interpret.

Reply:
Do you mean exceptions or logs here? try/catch won't affect logs obviously.

A try / catch would just presumably wrap the mongo exception inside a different exception with a message like "Failed to connect to db" which is why I didn't add it - it doesn't seem very useful and winds up as a 500 at the API level regardless

If we find that it helps to wrap the exception in the future I'm fine with adding it but right now I'm not seeing much of a benefit